telosnetwork/telos-wallet
# Enhancing EVM data retrieval (hyperion, data parsing, new types)

# Fixes #205 and #203

## Description
The current state of the new library lacks support for retrieving data from EVM which is a crucial feature of this new lib. This PR includes the first approach to that support by incorporating all useful javascript code from Teloscan that currently solves that problem but migrated to a fully typed typescript code.

This PR also includes the Home page made by @EJSG, so you already can log in using your Metamask Wallet from the new interface. Unfortunately, the new layout for the evm user is not finished, so this PR includes a temporal placeholder page to show the logged account address and a logout button to go back to Homepage.

## Test Scenarios
- go to [this page](https://deploy-preview-219--wallet-staging.netlify.app/)
  - You should see the new Homepage interface (@EJSG's [great work](https://github.com/telosnetwork/telos-wallet/pull/193))
  - "Telos EVM" should be selected by default
- Press the "Connect your Wallet" Button
  - Metamask wallet should pop up to prompt you to add the chain (if not present) and then, prompt you to log in.
- If you login
  - you should be redirected to a temporal URL `/evm-balance` with a temporal page
  - you should see the account address and a "logout" button.
- Press the logout button
  - you should be redirected to the homepage

## Checklist:
